docs/explanation.rst

```diff
@@ -316,6 +316,189 @@ are meant to be omitted from reading or have their time adjusted. We recommend
+
+    \chi_G^2 = n^T N n + b^T B^{-1} b
+
+.. _Coles_et_al_2011: https://ui.adsabs.harvard.edu/abs/2011MNRAS.418..561C/abstract
+.. _Lentati_et_al_2013: https://ui.adsabs.harvard.edu/abs/2013PhRvD..87j4021L/abstract
+.. _van_Haasteren_and_Vallisneri_2015: https://ui.adsabs.harvard.edu/abs/2015MNRAS.446.1170V/abstract
+.. _NANOGrav_9-year: https://ui.adsabs.harvard.edu/abs/2015ApJ...813...65N/abstract
+
+Fitting algorithms
+''''''''''''''''''
+
+PINT is designed to be able to offer several alternative algorithms to arrive
+at the best-fit model. This both because fitting can be a time-consuming
+process if a suboptimal algorithm is chosen, and because different kinds of
+model and data require different calculations - narrowband (TOA-only) versus
+wideband (TOA and DM measurements) and uncorrelated errors versus correlated
+errors.
+
+The TEMPO/TEMPO2 and default PINT fitting algorithms (:class:`pint.fitter.WidebandTOAFitter` for example), leaving aside the rank-reduced case, proceed like:
+
+1. Evaluate the model and its derivatives at the starting point :math:`x`, producing a set of residuals :math:`\delta y` and a Jacobian `M`.
+2. Compute :math:`\delta x` to minimize :math:`\left| M\delta x - \delta y \right|_C`, where :math:`\left| \cdot \right|_C` is the squared amplitude of a vector with respect to the data uncertainties/covariance :math:`C`.
+3. Update the starting point by :math:`\delta x`.
+
+TEMPO and TEMPO2 can check whether the predicted improvement of chi-squared, assuming the linear model is correct, is enough to warrant continuing; if so, they jump back to step 1 unless the maximum number of iterations is reached. PINT does not contain this check.
+
+This algorithm is the Gauss-Newton_algorithm_ for solving nonlinear
+least-squares problems, and even in one-complex-dimensional cases can exhibit
+convergence behaviour that is literally chaotic_. For TEMPO/TEMPO2 and PINT, the
+problem is that the model is never actually evaluated at the updated starting
+point before committing to it; it can be invalid (ECC > 1) or the step can be
+large enough that the derivative does not match the function and thus the
+chi-squared value after the step can be worse than the initial chi-squared.
+These issues particularly arise with poorly constrained parameters like M2 or
+SINI. Users experienced with pulsar timing are frequently all too familiar with
+this phenomenon and have a collection of tricks for evading it.
+
+PINT contains a slightly more sophisticated algorithm, implemented in
+:class:`pint.fitter.DownhillFitter`, that takes more careful steps:
+
+1. Evaluate the model and its derivatives at the starting point :math:`x`, producing a set of residuals :math:`\delta y` and a Jacobian `M`.
+2. Compute :math:`\delta x` to minimize :math:`\left| M\delta x - \delta y \right|_C`, where :math:`\left| \cdot \right|_C` is the squared amplitude of a vector with respect to the data uncertainties/covariance :math:`C`.
+3. Set :math:`\lambda` to 1.
```

Comment:
This renders wrong. I think it should be 4. then 5. for the next line.

Reply:
Oops. Quite right!